### PR TITLE
chore Update native library typings with dts-generator tool

### PR DESCRIPTION
**Pull request by VladimirAmiorkov**
chore: Change deprecated 'compile' with 'implementation'

Merge only after [tns-platform-declarations](https://www.npmjs.com/package/tns-platform-declarations) has released a new version with the expected typings.

## PR Checklist

- [X] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [ ] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [X] All existing tests are passing
- [ ] Tests for the changes are included

## What is the new behavior?

Fixes/Implements/Closes #[Issue Number].

BREAKING CHANGES:

Raises the lowest supported level by the plugin to API 21 (This is shown by the generated d.ts but should be tested if the 1.9.0 native lib actually works on lower API levels)

### CODE REVIEW

**Comment by VladimirAmiorkov** [comment]
run ci

File: demo-angular/package.json [hunk 1]
```diff
@@ -30,16 +30,21 @@
   },
   "devDependencies": {
     "htmlparser2": "^3.9.2",
+    "@angular/compiler-cli": "~6.0.0",
```
Comment (lini) [inline demo-angular/package.json]: I think the changes to this file are not needed and it should not be committed.

File: demo/package.json [hunk 2]
```diff
@@ -16,11 +16,15 @@
     "tns-core-modules": "^4.2.0"
   },
   "devDependencies": {
-    "nativescript-dev-typescript": "^0.7.2",
-    "nativescript-dev-webpack": "~0.15.1",
-    "typescript": "~2.8.2",
+    "babel-traverse": "6.26.0",
```
Comment (lini) [inline demo/package.json]: I think the changes to this file are not needed and it should not be committed.

**Review by lini — CHANGES_REQUESTED** [review]

File: demo-angular/package.json (hunk 1 above)
Comment (VladimirAmiorkov) [reply demo-angular/package.json]: Reverted to original but added the `@angular/compiler-cli` and `@ngtools/webpack` `devDependencies` that are automatically added by the new webpack

File: demo/package.json (hunk 2 above)
Comment (VladimirAmiorkov) [reply demo/package.json]: Reverted to original

**Review by lini — APPROVED** [review]

**Pull request merged**